orienteer-pivottable/src/main/java/org/orienteer/pivottable/component/widget/BrowsePivotTableWidget.java

```diff
@@ -56,7 +59,7 @@ public BrowsePivotTableWidget(String id, IModel<OClass> model,
 	}
 
 	public String getUrl() {
-		String sql = getSql();
+		String sql =  !Strings.isEmpty(customSQL) ? customSQL : getSql();
```

Comment:
As commented previously: why just do not incupsulate everything to "getSql()" method?